eng/scripts/Compile-Changelog.ps1

```diff
@@ -0,0 +1,711 @@
+#!/usr/bin/env pwsh
+<#
+.SYNOPSIS
+    Compiles changelog entries from YAML files into CHANGELOG.md.
+
+.DESCRIPTION
+    This script reads all YAML files from the changelog-entries directory,
+    validates them against the schema, groups them by section and subsection,
+    and inserts the compiled entries into CHANGELOG.md under the specified version section.
+    
+    If no version is specified, entries are added to the "Unreleased" section at the top.
+    If there is no "Unreleased" section and no version is specified, a new "Unreleased" 
+    section is created using the next semantic version number.
+
+.PARAMETER ServerName
+    Name of the server to compile changelog for (e.g., "Azure.Mcp.Server", "Fabric.Mcp.Server").
+    Defaults to "Azure.Mcp.Server".
+
+.PARAMETER ChangelogPath
+    Path to the CHANGELOG.md file. If not specified, uses servers/{ServerName}/CHANGELOG.md.
+
+.PARAMETER ChangelogEntriesPath
+    Path to the changelog-entries directory. If not specified, uses servers/{ServerName}/changelog-entries.
+
+.PARAMETER Version
+    Target version section to compile entries into (e.g., "2.0.0-beta.3", "1.5.2").
+    If not specified, uses the "Unreleased" section or creates one.
+
+.PARAMETER DryRun
+    Preview what will be compiled without modifying any files.
+
+.PARAMETER DeleteFiles
+    Delete YAML files after successful compilation.
+
+.EXAMPLE
+    ./eng/scripts/Compile-Changelog.ps1 -DryRun
+
+    Preview what will be compiled for Azure.Mcp.Server without making changes.
+
+.EXAMPLE
+    ./eng/scripts/Compile-Changelog.ps1 -ServerName "Fabric.Mcp.Server"
+
+    Compile entries into the Unreleased section for Fabric.Mcp.Server.
+
+.EXAMPLE
+    ./eng/scripts/Compile-Changelog.ps1 -Version "2.0.0-beta.3"
+
+    Compile entries into the 2.0.0-beta.3 version section for Azure.Mcp.Server.
+
+.EXAMPLE
+    ./eng/scripts/Compile-Changelog.ps1 -ServerName "Fabric.Mcp.Server" -Version "1.0.0"
+
+    Compile entries into the 1.0.0 version section for Fabric.Mcp.Server.
+
+.EXAMPLE
+    ./eng/scripts/Compile-Changelog.ps1 -DeleteFiles
+
+    Compile entries and remove YAML files after successful compilation.
+#>
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $false)]
+    [string]$ServerName = "Azure.Mcp.Server",
```

Comment:
Like New-ChangelogEntry, we shouldn't favor any specific server in a repo-common script.